src/deserialization_main.hpp

```diff
@@ -0,0 +1,148 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
```

Comment:
what do you mean with _main? is it leftover that has not been split to frontends? is it remaining todo or it is final state